github_jira/github/github.go

```diff
@@ -0,0 +1,138 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v35/github"
+	"github.com/mattermost/mattermost-utilities/github_jira/jira"
+	"github.com/mattermost/mattermost-utilities/github_jira/verbiages"
+	"golang.org/x/oauth2"
+)
+
+type label struct {
+	Name string `json:"name"`
+}
+
+func has(needle string, haystack []string) bool {
+	for _, item := range haystack {
+		if needle == item {
+			return true
+		}
+	}
+	return false
+}
+
+func ParseRepo(repoStr string) (repo, error) {
+	ownerAndRepo := strings.Split(repoStr, "/")
+	if len(ownerAndRepo) != 2 {
+		return repo{}, errors.New(fmt.Sprintf(`Expected repo to be of form "<owner>/<repo>", but got %s`, repoStr))
+	}
+	return repo{
+		owner: ownerAndRepo[0],
+		repo:  ownerAndRepo[1],
+	}, nil
+}
+
+type repo struct {
+	owner string
+	repo  string
+}
+
+func CreateIssues(jiraBasicAuth string, ghToken string, repo repo, labels []string, jiraIssues []jira.Issue, dryRun bool) string {
```

Comment:
This return value looks wrong, you are using `string` as an unstructured data that you pass back to other different place. This method is taking two responsabilities, creating the issues and formating the output. My expectation would be to return something like `(CreationResult, error)` where creation results have all the structured information needed. And the caller is going to be responsible of using that structured that to generate the output.